host/windows/syscall.c

```diff
@@ -382,18 +793,75 @@ oe_host_fd_t oe_syscall_open_ocall(
 
 ssize_t oe_syscall_read_ocall(oe_host_fd_t fd, void* buf, size_t count)
 {
-    if ((count & UINT_MAX) != count)
-        _set_errno(OE_EINVAL);
+    ssize_t ret = -1;
+    DWORD bytes_returned = 0;
+
+    // Convert fd 0, 1, 2 as needed
+    switch (fd)
+    {
+        case 0:
+            fd = (oe_host_fd_t)GetStdHandle(STD_INPUT_HANDLE);
+            break;
+
+        case 1:
+            _set_errno(OE_EBADF);
+            goto done;
+
+        case 2:
+            _set_errno(OE_EBADF);
+            goto done;
+
+        default:
+            break;
+    }
+
+    if (!ReadFile((HANDLE)fd, buf, (DWORD)count, &bytes_returned, NULL))
```

Comment:
Suggestion: This logic can probably be moved to case 0, but I might be mistaken?